src/expanduser.cpp

```diff
@@ -133,22 +133,47 @@ bool IsNetworked(unsigned long ft)
 #undef CASE_RET_STRING
 #endif // __linux__
 
-bool IsNetworkedFilesystem(const boost::filesystem::path& path)
+bool IsNetworkedFilesystem(const boost::filesystem::path& path_)
 {
 #ifdef __linux__
+    // Non-DEV builds put user databases in ~/.config/miopen by default; the binary cache is placed
+    // in ~/.cache/miopen. If these directories do not exist, this is not a problem, because the
+    // library creates them as needed.
+    //
+    // The problem is that statfs doesn't work in this case, and we need to determine the type of FS
+    // _before_ the databases are created.
+    //
+    // Solution (A): Just create_directories if path doesn't exist. It looks like a hack: if the
+    // path is on NFS, then the library will not use it, but will create directories (which is not
+    // good). And this function should not have any side effects on the file system.
+    //
+    // Solution (B): Traverse the input path up to the first existing directory, then check that
+    // directory. Stop after some fixed number of iterations to protect against possible file system
+    // problems. Let's use this solution for now.
+    auto path = path_;
+    for(int i = 0; i < 32; ++i)
```

Comment:
1. Does this 32 denote the number of attempts to connect to the network filesystem? Please promote it to a macro with a name denoting this.

2. If this performs repeated attempts, should not a delay be added between them?

Reply:
1. The source code says:
```C++
// Stop after some fixed number of iterations to protect against possible file system problems.
```
and `32` is used immediately after that. So the intent should be clear for the reader. Symbolic representation of this literal as a macro won't provide additional clarity to the text.

And we use this literal only once. Therefore for now I do not see enough reasons for adding a macro.

Will do if you insist.

2. I do not think so. What the code does it _not_ repeated attempts to do the same action.